models/product-type/src/builder.spec.ts

```diff
@@ -0,0 +1,124 @@
+              de: expect.any(String),
+              en: expect.any(String),
+              fr: expect.any(String),
+            }),
+            isRequired: expect.any(Boolean),
+            attributeConstraint: expect.any(String),
+            inputTip: expect.objectContaining({
+              de: expect.any(String),
+              en: expect.any(String),
+              fr: expect.any(String),
+            }),
+            inputHint: expect.any(String),
+            isSearchable: expect.any(Boolean),
+          }),
+        ]),
+      })
+    )
+  );
+
+  it(
+    ...createBuilderSpec<TProductType, TProductType>(
+      'rest',
+      ProductType.random(),
+      expect.objectContaining({
+        id: expect.any(String),
+        key: expect.any(String),
+        version: expect.any(Number),
+        createdAt: expect.any(String),
+        createdBy: expect.objectContaining({
+          customer: expect.objectContaining({ typeId: 'customer' }),
+        }),
+        lastModifiedAt: expect.any(String),
+        lastModifiedBy: expect.objectContaining({
+          customer: expect.objectContaining({ typeId: 'customer' }),
+        }),
+        name: expect.any(String),
+        description: expect.any(String),
+        attributes: expect.arrayContaining([
+          expect.objectContaining({
+            type: expect.objectContaining({ name: expect.any(String) }),
+            name: expect.any(String),
+            label: expect.objectContaining({
+              de: expect.any(String),
+              en: expect.any(String),
+              fr: expect.any(String),
+            }),
+            isRequired: expect.any(Boolean),
+            attributeConstraint: expect.any(String),
+            inputTip: expect.objectContaining({
+              de: expect.any(String),
+              en: expect.any(String),
+              fr: expect.any(String),
+            }),
+            inputHint: expect.any(String),
+            isSearchable: expect.any(Boolean),
+          }),
+        ]),
+      })
+    )
+  );
+
+  it(
+    ...createBuilderSpec<TProductType, TProductTypeGraphql>(
+      'graphql',
+      ProductType.random(),
+      expect.objectContaining({
+        __typename: 'ProductTypeDefinition',
+        id: expect.any(String),
+        key: expect.any(String),
+        version: expect.any(Number),
+        createdAt: expect.any(String),
+        createdBy: expect.objectContaining({
+          customerRef: expect.objectContaining({ typeId: 'customer' }),
+          userRef: expect.objectContaining({ typeId: 'user' }),
+        }),
+        lastModifiedAt: expect.any(String),
+        lastModifiedBy: expect.objectContaining({
+          customerRef: expect.objectContaining({ typeId: 'customer' }),
+          userRef: expect.objectContaining({ typeId: 'user' }),
+        }),
+        name: expect.any(String),
+        description: expect.any(String),
+        // Gets error: Builder with name 'buildGraphql' does not exist on field 'attributeDefinitions'.
```

Comment:
This is because you don't need to put `attributeDefinitions` in the `buildFields` array. That is needed when it's a separate model you are pulling in. In this case, `attributeDefinitions` is a straight object.

Reply:
grr. I knew that, until I didn't.

Reply:
Fixed with [58b8863](https://github.com/commercetools/test-data/pull/198/commits/58b886309142a533898537cd95167277ec589c8d)